src/Sparrow.Server/Utils/DiskStatsGetter.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using Sparrow.Logging;
+using Sparrow.Server.Platform;
+
+namespace Sparrow.Server.Utils
+{
+    public interface IDiskStatsGetter
+    {
+        DiskStatsResult Get(string drive);
+    }
+
+    internal class DiskStatsGetter : IDiskStatsGetter
+    {
+        private readonly double _minInterval;
+        private static readonly Logger Logger = LoggingSource.Instance.GetLogger("Server", typeof(DiskStatsGetter).FullName);
+
+        private readonly object _locker = new object();
+        private readonly Dictionary<string, PrevInfo> _previousInfo = new Dictionary<string, PrevInfo>();
+
+        public DiskStatsGetter(TimeSpan minInterval)
+        {
+            _minInterval = minInterval.TotalSeconds;
+        }
+        
+        public DiskStatsResult Get(string drive)
+        {
+            if (drive == null)
+                return null;
+
+            lock (_locker)
+            {
+                if (_previousInfo.TryGetValue(drive, out var prev))
+                {
+                    var diff = (DateTime.UtcNow - prev.Raw.Time).TotalSeconds;
+                    if (diff < _minInterval)
+                        return prev.Calculated;
+
+                    var currentInfo = GetDiskInfo(drive);
```

Comment:
You are doing I/O under the lock. Who might contend with you here?

Reply:
Are you asking why there is a need for the lock at all or why the `GetDiskInfo` is in the lock?
I wanted to check first the time diff to avoid getting the stats if no need.

Reply:
Yes, the problem is that two accesses for this will lock, and can cause convoy.
Better to use `ConcurrentDictionary` and a `Task` instead.